### ISSUE TITLE
[Enhance] Button component should be triggerable by Space and or Enter - Angular

### ISSUE THREAD

**Issue by ssabo1** [sparkdesignsystem/spark-design-system#2783]
**User Story:**
As Spark we want our Button component to be triggerable by Space or Enter so that it is accessible.

**Notes:**
(This came up in the Paciello audit.)

**AC:**
Buttons can appear as either a <button> or as a <a>. When they appear as a <a> in React or Angular, they need to 
1. appear in the tab order
2. be triggerable with the Space and Enter keys.
3. Space should not scroll the page
